src/VkBootstrap.cpp

```diff
@@ -16,18 +16,165 @@
+		get_inst_proc_addr (fp_vkEnumerateInstanceVersion, "vkEnumerateInstanceVersion");
+		get_inst_proc_addr (fp_vkCreateInstance, "vkCreateInstance");
+	}
+
+	void init_instance_funcs (VkInstance inst) {
+		instance = inst;
+		get_inst_proc_addr (fp_vkDestroyInstance, "vkDestroyInstance");
+		get_inst_proc_addr (fp_vkEnumeratePhysicalDevices, "vkEnumeratePhysicalDevices");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceFeatures, "vkGetPhysicalDeviceFeatures");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceFeatures2, "vkGetPhysicalDeviceFeatures2");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceFormatProperties, "vkGetPhysicalDeviceFormatProperties");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceImageFormatProperties, "vkGetPhysicalDeviceImageFormatProperties");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceProperties, "vkGetPhysicalDeviceProperties");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceProperties2, "vkGetPhysicalDeviceProperties2");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceQueueFamilyProperties, "vkGetPhysicalDeviceQueueFamilyProperties");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceQueueFamilyProperties2, "vkGetPhysicalDeviceQueueFamilyProperties2");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceMemoryProperties, "vkGetPhysicalDeviceMemoryProperties");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceFormatProperties2, "vkGetPhysicalDeviceFormatProperties2");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceMemoryProperties2, "vkGetPhysicalDeviceMemoryProperties2");
+
+		get_inst_proc_addr (fp_vkCreateDevice, "vkCreateDevice");
+		get_inst_proc_addr (fp_vkDestroyDevice, "vkDestroyDevice");
+		get_inst_proc_addr (fp_vkEnumerateDeviceExtensionProperties, "vkEnumerateDeviceExtensionProperties");
+		get_inst_proc_addr (fp_vkGetDeviceQueue, "vkGetDeviceQueue");
+
+		get_inst_proc_addr (fp_vkDestroySurfaceKHR, "vkDestroySurfaceKHR");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceSurfaceSupportKHR, "vkGetPhysicalDeviceSurfaceSupportKHR");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceSurfaceFormatsKHR, "vkGetPhysicalDeviceSurfaceFormatsKHR");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceSurfacePresentModesKHR, "vkGetPhysicalDeviceSurfacePresentModesKHR");
+		get_inst_proc_addr (fp_vkGetPhysicalDeviceSurfaceCapabilitiesKHR, "vkGetPhysicalDeviceSurfaceCapabilitiesKHR");
+		get_inst_proc_addr (fp_vkCreateSwapchainKHR, "vkCreateSwapchainKHR");
+		get_inst_proc_addr (fp_vkDestroySwapchainKHR, "vkDestroySwapchainKHR");
+		get_inst_proc_addr (fp_vkGetSwapchainImagesKHR, "vkGetSwapchainImagesKHR");
+		get_inst_proc_addr (fp_vkCreateImageView, "vkCreateImageView");
+		get_inst_proc_addr (fp_vkDestroyImageView, "vkDestroyImageView");
+	}
+
+
+	bool load_vulkan (PFN_vkGetInstanceProcAddr fp_vkGetInstanceProcAddr = nullptr) {
+		if (fp_vkGetInstanceProcAddr != nullptr) {
+			ptr_vkGetInstanceProcAddr = fp_vkGetInstanceProcAddr;
+			return true;
+		} else {
+			auto lib = get_vulkan_library ();
+			ptr_vkGetInstanceProcAddr = lib.ptr_vkGetInstanceProcAddr;
```

Comment:
This function is almost thread-safe. If this function is thread-unsafe it makes everything invisibly thread-unsafe above.

Reply:
Solved with a mutex.